### PR TITLE
Prof: Logged in user update password

### PR DESCRIPTION
**Pull request by chidioguejiofor**
#### What does this PR achieve?
- improves error handling in Form
- extends Form component to handle image errors
- adds validation for form added

#### Manual Testing Steps
- goto /dashboard/password
- change your password

#### Screenshots:
Screenshot 2020-02-10 at 2 09 31 AM
Screenshot 2020-02-10 at 2 09 13 AM
Screenshot 2020-02-10 at 2 09 04 AM
Screenshot 2020-02-10 at 2 08 31 AM

#### Relevant Trello Tickets
- [User Update Passwird](https://trello.com/c/QCQX3eWL/54-d-2-user-should-be-able-to-update-their-password-from-their-dashboard)

### CODE REVIEW

File: src/components/pages/UpdatePassword/index.tsx [hunk 1]
```diff
@@ -1,8 +1,108 @@
...
+    newPassword: '',
+    currentPassword: '',
+    confirmPassword: '',
+  };
+  function isAnyInputBoxEmpty(composedProps) {
+    const { values } = composedProps;
+    const inputVals = [values.newPassword, values.currentPassword, values.confirmPassword];
+
+    return inputVals.some(value => value === '');
+  }
```
Comment (chukwuemekachm) [inline src/components/pages/UpdatePassword/index.tsx:36]: This functionality has now been duplicated across features. Can we put it in the form so as to reuse it?

File: src/components/pages/UpdateProfile/index.tsx [hunk 2]
```diff
@@ -17,21 +17,41 @@ interface UpdateProfileProps extends DashboardProps {
...
+      returnedErrors = formErrors;
+    } else {
+      returnedErrors = error.errors;
+    }
+    // if(Object.keys(error.errors).length){
+    //    return error.errors
+    // }
+    // console.log(error);
+
+    console.log(returnedErrors);
```
Comment (chukwuemekachm) [inline src/components/pages/UpdateProfile/index.tsx]: Can we tidy this?

**Review by chukwuemekachm — CHANGES_REQUESTED** [review]
Nice work @chidioguejiofor .

Just a few concerns.

File: src/components/pages/UpdateProfile/index.tsx (hunk 2 above)
Comment (chidioguejiofor) [reply src/components/pages/UpdateProfile/index.tsx]: 🤦🏼‍♀️Oversight

Comment (chidioguejiofor) [reply src/components/pages/UpdateProfile/index.tsx]: Thanks for that

File: src/components/pages/UpdatePassword/index.tsx (hunk 1 above)
Comment (chidioguejiofor) [reply src/components/pages/UpdatePassword/index.tsx:36]: Does not work in exactly the same way. Here the submit button is enabled when values have been inputted in all the text fields while in the `UpdateProfile` submit button is enabled when you input a new value(different from your current profile data) for _any_ of the fields. 

One more thing, I suggest that we actually repeat say 3-5 times before abstracting logic. That way we have enough examples and similarities of the different occurrences when abstracting

Comment (chukwuemekachm) [reply src/components/pages/UpdatePassword/index.tsx:36]: Great.

**Pull request merged**